pandas/core/internals.py

```diff
@@ -3486,10 +3486,20 @@ def replace_list(self, src_list, dest_list, inplace=False, regex=False,
         # figure out our mask a-priori to avoid repeated replacements
         values = self.as_matrix()
 
-        def comp(s):
-            if isna(s):
-                return isna(values)
-            return _maybe_compare(values, getattr(s, 'asm8', s), operator.eq)
+        if is_datetime64tz_dtype(self):
```

Comment:
this is not the right place to fix if this is even an issue

Reply:
I tend to agree, could use advice on where _is_ the right place.  Without this change there is a test failure in tests.indexing.test_coercion.TestReplaceSeriesCoercion.test_replace_series_datetime64tz with traceback

```
    def test_replace_series_datetime64tz(self):
        from_key = 'datetime64[ns, US/Eastern]'
        for to_key in self.rep:
>           self._assert_replace_conversion(from_key, to_key, how='dict')

pandas/tests/indexing/test_coercion.py:1317: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
pandas/tests/indexing/test_coercion.py:1239: in _assert_replace_conversion
    result = obj.replace(replacer)
pandas/core/generic.py:4514: in replace
    limit=limit, regex=regex)
pandas/core/generic.py:4563: in replace
    regex=regex)
pandas/core/internals.py:3504: in replace_list
    masks = [comp(s) for i, s in enumerate(src_list)]
pandas/core/internals.py:3502: in comp
    operator.eq)
pandas/core/internals.py:4954: in _maybe_compare
    result = op(a, b)
pandas/core/indexes/datetimes.py:122: in wrapper
    self._assert_tzawareness_compat(other)
```

where the relevant `obj.replace(replacer)` call has

```
index = pd.Index([3, 4], name='xxx')
data = [Timestamp('2011-01-01 00:00:00-0500', tz='US/Eastern'), Timestamp('2011-01-03 00:00:00-0500', tz='US/Eastern')]
obj = pd.Series(data, index=index, name='yyy')
replacer = {Timestamp('2011-01-01 00:00:00-0500', tz='US/Eastern'): 1.1, Timestamp('2011-01-03 00:00:00-0500', tz='US/Eastern'): 2.2}
```

The problem is that in internals `replace_list` is converting to m8 which drops tz.

Reply:
this is not the place for this, might need to fixe ``.replace`` in the block itself.